packages/modules-sdk/src/remote-link.ts

```diff
@@ -0,0 +1,379 @@
+              })
+
+              let cascadeDelKeys: DeleteEntities = {}
+              cascadeDelKeys[primaryKey] = unprocessedIds
+              const service: ILinkModule = this.modulesMap.get(serviceName)!
+
+              const returnFields = this.getLinkableKeys(
+                service as LoadedLinkModule
+              )
+
+              let deletedEntities: Record<string, string[]> = {}
+
+              try {
+                if (args?.methodSuffix) {
+                  method += toPascalCase(args.methodSuffix)
+                }
+
+                const [_, removed] = await service[method](cascadeDelKeys, {
+                  returnLinkableKeys: returnFields,
+                })
+
+                deletedEntities = removed as Record<string, string[]>
+              } catch (error) {
+                errors.push({
+                  serviceName,
+                  method,
+                  args: cascadeDelKeys,
+                  error: JSON.parse(
+                    JSON.stringify(error, Object.getOwnPropertyNames(error))
+                  ),
+                })
+                return
+              }
+
+              if (Object.keys(deletedEntities).length === 0) {
+                return
+              }
+
+              removedIds[serviceName] = {
+                ...deletedEntities,
+              }
+
+              if (!isCascading) {
+                returnIdsList[serviceName] = {
+                  ...deletedEntities,
+                }
+              } else {
+                const [mainKey] = returnFields
+
+                if (!returnIdsList[serviceName]) {
+                  returnIdsList[serviceName] = {}
+                }
+                if (!returnIdsList[serviceName][mainKey]) {
+                  returnIdsList[serviceName][mainKey] = []
+                }
+
+                returnIdsList[serviceName][mainKey] = [
+                  ...new Set(
+                    returnIdsList[serviceName][mainKey].concat(
+                      deletedEntities[mainKey]
+                    )
+                  ),
+                ]
+              }
+
+              Object.keys(deletedEntities).forEach((key) => {
+                deletedEntities[key].forEach((id) => {
+                  const hash = `${serviceName}-${key}`
+                  if (!processedIds[hash]) {
+                    processedIds[hash] = new Set()
+                  }
+
+                  processedIds[hash].add(id)
+                })
+              })
+
+              await cascade(
+                [
+                  {
+                    serviceName: serviceName,
+                    deleteKeys: deletedEntities as DeleteEntities,
+                  },
+                ],
+                true
+              )
+            }
+          )
+
+          deletePromises.push(...relatedServicesPromises)
+        }
+
+        await Promise.all(deletePromises)
+      })
+
+      await Promise.all(servicePromises)
```

Comment:
question: I am wondering if having a triple promises array touching multiple modules can generate issues with the module internal transaction? My thinking is that until now, when a module is creating a transction internally and is used in a map, it will create X concurrent transction and can block the db if there is enough connections involved in the process and that a transaction is blocked by another one. So we might need to test this deeply when we are there. are you ok?